### PR TITLE
Enable automatically collect optimized latency in NanoHPO

### PR DESCRIPTION
**Pull request by rnwang04**
## Description
This PR want to enable automatic latency collection for different accelerated model (e.g. ONNXRuntime, OpenVINO, JIT_IPEX) in NanoHPO and return the accelerated minimum latency to user.

### 1. Why the change?

Help user get the accelerated minimum latency when using NanoHPO.

### 2. User API changes
In NanoHPO
```
trials = trainer.search(
            model,
            target_metric=['val_loss', 'latency'],  # latency is a built-in metric
            directions=['minimize', 'minimize'],
            n_trials=4,
            max_epochs=3,
            acceleration=True,
            input_sample=input_sample
        )
```
Muti-Objective HPO in BaseForecaster of Chronos
```
trials = forecaster.tune(data=train_data, validation_data=val_data,
                target_metric=['mse','latency'],
                direction=None,
                directions=['minimize', 'minimize'],
                epochs=3,
                acceleration=True)
```
in which `acceleration` and `input_sample` are new parameters.
### 3. Summary of the change 
- add _help.py to refactor code
- fix a bug of PytorchIPEXJITModel and add forward_args property for IPEXJITModel
- add _auto_optimize func for Objective to enable automatic latency collection for different accelerated model on the end of each trial (e.g. ONNXRuntime, OpenVINO, JIT_IPEX) 
- add unittest in NanoHPO for acceleration option
- Migrated multi-target search and built-in acceleration to BaseForecaster in Chronos, fixed some bugs in the migration process
- Migrated multi-target search and built-in acceleration to AutoformerForecaster in Chronos, fixed some bugs in the migration process
- Support seed search in BaseForecaster
- add unittest in TCNForecaster for multi-target search and acceleration option
- add unittest in AutoformerForecaster for multi-target search and acceleration option

### 4. How to test?
- [x] Unit test
- [x] Jenkins
http://10.112.231.51:18888/job/BigDL-NanoTests-PR-Validation/342/
http://10.112.231.51:18889/job/BigDL-Chronos-PR-Validation/650/

### CODE REVIEW

File: python/nano/src/bigdl/nano/automl/pytorch/objective.py [hunk 1]
```diff
@@ -183,19 +147,57 @@ def _fix_data_args(self,
...
+
+        # here we suppose nn.model is attached to plmodel by attribute "model"
+        original_model = model.model
+        for optimization in ['openvino', 'onnxruntime', 'jit']:
+            # may enable more optimizations later
+            try:
+                if optimization == 'jit':
+                    optim_model = Trainer.trace(original_model, input_sample=self.input_sample,
+                                                accelerator=optimization,
+                                                use_ipex=True, channels_last=False)
```
Comment (TheaperDeng) [inline python/nano/src/bigdl/nano/automl/pytorch/objective.py:163]: I am not quite sure if states `channels_last=False` by default is correct.

File: python/nano/src/bigdl/nano/automl/pytorch/objective.py [hunk 2]
```diff
@@ -183,19 +147,57 @@ def _fix_data_args(self,
...
+            for metric in self.target_metric:
+                score = self.searcher.trainer.callback_metrics[metric].item()
+                optim_score.append(score)
+            # compare optim_scores with original scores to find a similar
+            # loss value with less latency
+            optim_score = Score(*optim_score)
+            usable = True
+            for metric in self.target_metric:
+                if metric != "latency":
+                    if abs(getattr(optim_score, metric) - getattr(best_score, metric)) >= 0.005:
```
Comment (TheaperDeng) [inline python/nano/src/bigdl/nano/automl/pytorch/objective.py:182]: use proportion here

File: python/nano/src/bigdl/nano/automl/pytorch/objective.py [hunk 3]
```diff
@@ -183,19 +147,57 @@ def _fix_data_args(self,
...
+            model.model = optim_model
+            self.searcher._validate(model, self.val_dataloaders)
+            optim_score = []
+            for metric in self.target_metric:
+                score = self.searcher.trainer.callback_metrics[metric].item()
+                optim_score.append(score)
+            # compare optim_scores with original scores to find a similar
+            # loss value with less latency
+            optim_score = Score(*optim_score)
+            usable = True
```
Comment (TheaperDeng) [inline python/nano/src/bigdl/nano/automl/pytorch/objective.py:179]: add some comment to explain why this usable is needed (for future developer)

File: python/nano/src/bigdl/nano/automl/pytorch/hposearcher.py [hunk 4]
```diff
@@ -124,6 +114,8 @@ def search(self,
                resume=False,
                target_metric=None,
                n_parallels=1,
+               auto_optimize=False,
```
Comment (TheaperDeng) [inline python/nano/src/bigdl/nano/automl/pytorch/hposearcher.py:117]: leave `auto_optimize=True`. it will be set to False by default if "latency" is not in target_metric

File: python/nano/src/bigdl/nano/automl/pytorch/hposearcher.py [hunk 5]
```diff
@@ -133,6 +125,11 @@ def search(self,
             defaults to False.
         :param target_metric: the object metric to optimize,
             defaults to None.
+        :param auto_optimize: Whether to automatically consider the model after
+            inference acceleration in the search process. It will only take
+            effect if target_metric contains "latency". Default value is False.
+        :param input_sample: A set of inputs for trace, defaults to None if you have
+            trace before or model is a LightningModule with any dataloader attached.
```
Comment (TheaperDeng) [inline python/nano/src/bigdl/nano/automl/pytorch/hposearcher.py:132]: add similar api doc in `Trainer.search` as will

File: python/nano/src/bigdl/nano/automl/pytorch/objective.py (hunk 1 above)
Comment (rnwang04) [reply python/nano/src/bigdl/nano/automl/pytorch/objective.py:163]: Do you think here use a try-catch to capture "channels_last=True" error is better or export channels_last parameter to user is better?

File: python/nano/src/bigdl/nano/automl/pytorch/hposearcher.py (hunk 4 above)
Comment (rnwang04) [reply python/nano/src/bigdl/nano/automl/pytorch/hposearcher.py:117]: If leave auto_optimize=True, we suppose if user use "latency" it will automatically trigger optimizations and of course it will be slower.Does It really should be a default option?

File: python/nano/src/bigdl/nano/automl/pytorch/hposearcher.py (hunk 5 above)
Comment (rnwang04) [reply python/nano/src/bigdl/nano/automl/pytorch/hposearcher.py:132]: Good catch, thx. Done.

File: python/nano/src/bigdl/nano/automl/pytorch/objective.py (hunk 2 above)
Comment (rnwang04) [reply python/nano/src/bigdl/nano/automl/pytorch/objective.py:182]: Good catch, thx. Done.

File: python/nano/src/bigdl/nano/automl/pytorch/objective.py (hunk 3 above)
Comment (rnwang04) [reply python/nano/src/bigdl/nano/automl/pytorch/objective.py:179]: done.

File: python/nano/src/bigdl/nano/pytorch/lightning.py [hunk 6]
```diff
@@ -53,10 +53,17 @@ def __init__(self, model: nn.Module, loss: _Loss = None, optimizer: Optimizer =
...
     @property
     def forward_args(self):  # noqa
-        return inspect.getfullargspec(self.model.forward).args[1:]
+        if self._forward_args_ is None:
+            self._forward_args_ = inspect.getfullargspec(self.model.forward).args[1:]
+        return self._forward_args_
+
+    @forward_args.setter
+    def forward_args(self, forward_args):
+        self._forward_args_ = forward_args
```
Comment (zhentaocc) [inline python/nano/src/bigdl/nano/pytorch/lightning.py:66]: What's this for?

Comment (rnwang04) [reply python/nano/src/bigdl/nano/pytorch/lightning.py:66]: To make forward_args editable, and for PytorchOpenVINOModel and similar models, I can pass model.forward_args to lightning.forward_args.
